platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/MigrationReport.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.util.Collection;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+/**
+ * The migration report provides information about the execution of a migration operation.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface MigrationReport {
```

Comment:
❗️  In the catalog core API, we often will provide property maps so we have a dynamic source of information transportation without impacting the API itself. Would it be a good idea to add a `Map<String, Object>` for report properties?